budcamp/Homework3/Homework3/task2/TestRunner.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework3
+{
+    internal class TestRunner
+    {
+        private TestContainer testContainer;
+        private TestResults testResults;
+
+        int correctAnswers = 0;
+        string userAnswer = "";
+
+        public TestRunner(TestContainer testContainer)
+        {
+            this.testContainer = testContainer;
+            testResults = new TestResults();
+        }
+
+        public void RunTest(string topic)
+        {
+            List<Test> chosenTopic = testContainer.GetTestByTopic(topic);
+
+            foreach (Test test in chosenTopic)
+            {
+                test.PrintTest();
+
+                if (test.CheckAnswer(userAnswer))
+                {
+                    correctAnswers++;
+                }
+            }
+
+            testResults.AddResults(topic, correctAnswers);
```

Comment:
Instead of calling `AddResults` here, you can remove this line and return `TestResult` at the end of the method, like so:
```
   TestResult result = new TestResult {
       Topic = topic,
       Tests = tests,
       Date = DateTime.Now,
       CorrectAnswers = correctAnswers
   };
   return result;
```